bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Pattern.java

```diff
@@ -85,14 +84,23 @@ public class Pattern extends Resource {
  */
 public Pattern(Device device, Image image) {
 	super(device);
+	isImagePattern = true;
 	if (image == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
 	if (image.isDisposed()) SWT.error(SWT.ERROR_INVALID_ARGUMENT);
 	this.device.checkGDIP();
-	long[] gdipImage = image.createGdipImage();
+	this.image = image;
+	initialZoom = DPIUtil.getDeviceZoom();
+	setImageHandle(image, initialZoom);
+	init();
+}
+
+void setImageHandle(Image image, int zoomLevel) {
```

Comment:
I would move that method below all constructors and it can be private, right?

Reply:
Done